radix-engine/src/vm/wasm/prepare.rs

```diff
@@ -983,15 +983,18 @@ impl WasmModule {
         mut self,
         rules: &R,
     ) -> Result<Self, PrepareError> {
-        let backend = gas_metering::host_function::Injector::new(
-            MODULE_ENV_NAME,
-            COSTING_CONSUME_WASM_EXECUTION_UNITS_FUNCTION_NAME,
-        );
-        gas_metering::inject(&mut self.module, backend, rules).map_err(|err| {
-            PrepareError::RejectedByInstructionMetering {
-                reason: err.to_string(),
-            }
-        })?;
+        #[cfg(not(feature = "coverage"))]
```

Comment:
Out of curiosity, did you run into any issue with gas metering ingestion for the coverage-enabled wasm code?

Reply:
Yes. When transaction fails for example due to out of gas then dump_coverage cannot be called because it will fail due to  out of gas. When this is commented, the function dump_coverage is not using any build-in native functions so it's going to work even in case of transaction failure.

Reply:
That makes sense. I was hoping it wasn't an error with the instrumentation itself, otherwise we have a problem. All is good.